cpp/src/io/csv/csv_writer.cu

```diff
@@ -50,23 +51,51 @@ NVStrings* column_to_strings_csv(const gdf_column* column, gdf_size_type row_off
     auto valid = column->valid;
     if( valid )                                    // normalize row_offset (number of bits here)
         valid += (row_offset / GDF_VALID_BITSIZE); // to appropriate pointer for the bitmask
+
+    auto gdf2nvs_timeunit = [] (gdf_time_unit time_unit) -> NVStrings::timestamp_units {
+        if( time_unit==TIME_UNIT_s )
+            return NVStrings::seconds;
+        if( time_unit==TIME_UNIT_us )
+            return NVStrings::us;
+        if( time_unit==TIME_UNIT_ns )
+            return NVStrings::ns;
+        return NVStrings::ms;
+    };
+
     switch( column->dtype )
     {
         case GDF_STRING:
             rtn = (static_cast<NVStrings*>(column->data))->sublist(row_offset,row_offset+rows);
             break;
+        case GDF_STRING_CATEGORY:
+        {
+            NVCategory* category = reinterpret_cast<NVCategory*>(column->dtype_info.category);
+            if( category )
+                rtn = category->gather_strings((static_cast<const int32_t*>(column->data)) + row_offset, rows);
+            break;
+        }
         case GDF_BOOL8:
-            {
-            auto d_src = (static_cast<const cudf::bool8*>(column->data)) + row_offset;
-            device_buffer<bool> bool_buffer(rows);
-            thrust::transform(
-                rmm::exec_policy()->on(0), d_src, d_src + rows,
-                bool_buffer.data(),
-                [] __device__(const cudf::bool8 value) { return bool{value}; });
-            rtn = NVStrings::create_from_bools(bool_buffer.data(), rows,
+            rtn = NVStrings::create_from_bools((reinterpret_cast<const bool*>(column->data))+row_offset, rows,
```

Comment:
This isn't safe. There's no guarantee that `sizeof(bool) == sizeof(cudf::bool8)`.

Reply:
Yes, you are correct. Just looked that up. I will fix this.

Reply:
Could I add an if-statement for `sizeof(bool) != sizeof(cudf::bool8)`?
The transform is a waste if they are the same size.

Reply:
Hm, I suppose adding a `static_assert( sizeof(bool) == sizeof(cudf::bool8) )` is an acceptable solution for the short term.